griptape/drivers/vector/mongodb_vector_store_driver.py

```diff
@@ -80,11 +80,56 @@ def load_entries(
             )
 
     def query(
-        self,
-        query: str,
-        count: Optional[int] = None,
-        namespace: Optional[str] = None,
-        include_vectors: bool = False,
-        **kwargs
+            self,
+            query: str,
+            count: Optional[int] = None,
+            namespace: Optional[str] = None,
+            include_vectors: bool = False,
+            offset: Optional[int] = 0,
+            index: Optional[str] = None,
+            **kwargs
     ) -> list[BaseVectorStoreDriver.QueryResult]:
-        raise NotImplementedError("Vector search not supported in MongoDB")
+        collection = self.get_collection()
+
+        # Using the embedding driver to convert the query string into a vector
+        vector = self.embedding_driver.embed_string(query)
+
+        knn_k = count if count else 10
+        pipeline = [
+            {
+                "$search": {
+                    "index": index if index else "default",
```

Comment:
I was thinking of not setting it at all if it's not passed. You can just add a line after `pipeline = [...]` setting the key:

```python
if index:
    pipeline[0]["$search"]["index"] = index
```

Reply:
done